lib/db.py

```diff
@@ -23,43 +22,51 @@ def __init__(self):
             # force domain socket connection by not supplying host
             # pylint: disable=consider-using-f-string
             if config['postgresql']['host'] is None:
-                self._conn = psycopg.connect("user=%s dbname=%s password=%s port=%s" % (
-                    config['postgresql']['user'],
-                    config['postgresql']['dbname'],
-                    config['postgresql']['password'],
-                    config['postgresql']['port']))
+                self._pool = ConnectionPool(
+                    "postgresql://%s:%s@localhost:%s/%s" % (
+                        config['postgresql']['user'],
+                        config['postgresql']['password'],
+                        config['postgresql']['port'],
+                        config['postgresql']['dbname'],
+                    ),
+                    min_size=1,
+                    max_size=2,
```

Comment:
Why are you setting this to `max_size=2` Shouldn't this be the amount of threads the server can have? As threads share the obj?
```
/var/www/startup/venv/bin/gunicorn \
--workers=2 \
--access-logfile=- \
--error-logfile=- \
--worker-tmp-dir=/dev/shm \
--threads=4 \
--worker-class=gthread \
--bind unix:/tmp/green-coding-api.sock \
-m 007 \
--user www-data \
--chdir /var/www/green-metrics-tool/api \
-k uvicorn.workers.UvicornWorker \
main:app
```